src/System.Net.Http/src/System/Net/Http/Unix/CurlCallbacks.cs

```diff
@@ -414,5 +430,59 @@ private static void CheckForCompletedTransfers(SafeCurlMultiHandle multiHandle,
                 }
             }
         }
+
+        private static void PollFunction(SafeCurlMultiHandle multiHandle)
+        {
+            List<Interop.libc.pollfd> fds = new List<Interop.libc.pollfd>();
+
+            while (true)
+            {
+                lock (multiHandle)
+                {
+                    if (0 == multiHandle.FdCount)
+                    {
+                        multiHandle.PollCancelled = true;
+                        return;
+                    }
+                }
+
+                multiHandle.PollFds(fds);
+                if (0 == fds.Count)
+                {
+                    // ignore errors
```

Comment:
For now, I have changed this to throw so a tight loop should not happen. I need to think thru the error handling in the poll code path (have added a TODO for this)